lib/azure/armrest/model/base_model.rb

```diff
@@ -62,7 +74,20 @@ def initialize(json)
           @json = json
         end
 
-        @ostruct = OpenStruct.new(hash)
+        ostruct = AzureOpenStruct.new(hash)
+        keys = ostruct.to_h.keys
+
+        # Override the Delegate's normal behavior so that it any name collisions
+        # become methods for our model rather than skipped. An _alias is created
```

Comment:
This feels backwards...I would think you should let the original methods work as expected...and `_alias` methods to be the only entry point.  The other option is to just use a BasicObject instead, and then you don't have nearly as many collisions.